### ISSUE TITLE
Add MaximumOSVersion and/or SupportedOSVersions to manifest 

### ISSUE THREAD

**Issue by Trenly** [microsoft/winget-cli#2209]
### Description of the new feature / enhancement

The manifest currently allows for a MinimumOSVersion to distinguish apps which cannot be installed on older versions of the OS. However, there are also some packages (see microsoft/winget-pkgs#62490) which cannot be installed on newer versions of the OS such as Windows 11. This compatibility issue should be able to be placed in the manifest

### Proposed technical implementation details

1) Implement a field in the manifest for fully defining OS version support
1) Do something with that information
  * Hide the packages which are unsupported?
  * Mark the versions with a special character as suggested in #2134 ?
  * Display a warning that installing this package on a newer OS may not work, or the program may need to be run in compatibility mode?
  * Anything to keep users informed

**Comment by krystofkrticka** [comment]
I don't know about you but this package runs just fine on my Windows 11 installation. https://github.com/microsoft/winget-pkgs/pull/62490

**Comment by Trenly** [comment]
> I don't know about you but this package installs and runs just fine on my Windows 11 installation. [microsoft/winget-pkgs#62490](https://github.com/microsoft/winget-pkgs/pull/62490)

In the ShortDescription field of the manifest in that PR it says:
```yaml
ShortDescription: Spacedesk Driver Software for Windows PRIMARY PC (server) runs only with Windows 10, windows 8.1 and windows 7.
```

I didn't validate whether or not it actually ran or not. Either way, the fact still remains that the manifest currently has no concept of MaximumOSVersion

**Comment by krystofkrticka** [comment]
Oh I see that might be description, before Windows 11 was released because even on official website it says that it runs on Windows 10/11 this package in particular does not run on older version than Windows 10. I'll make another PR and fix it.

**Comment by krystofkrticka** [comment]
That's true. Especially windows terminal is annoying on Windows 10 when it show that newer version is available but that version is for Windows 11.
